lib/src/ui/puzzle/puzzle_dashboard_screen.dart

```diff
@@ -141,3 +147,80 @@ class _PuzzleButton extends StatelessWidget {
     );
   }
 }
+
+class DaysSelector extends ConsumerWidget {
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final session = ref.watch(userSessionStateProvider);
+    final day = ref.watch(daysProvider);
+    return session != null
+        ? InkWell(
+            onTap: () => showChoicesPicker(
+              context,
+              choices: Days.values,
+              selectedItem: day,
+              labelBuilder: (t) => Text(t.toString()),
+              onSelectedItemChanged: (newDay) {
+                ref.read(daysProvider.notifier).state = newDay;
+              },
+            ),
+            child: Row(
+              mainAxisAlignment: MainAxisAlignment.end,
+              children: [
+                Text(day.toString()),
+                const Icon(Icons.arrow_drop_down),
+              ],
+            ),
+          )
+        : const SizedBox.shrink();
+  }
+}
+
+enum Days {
+  oneday,
```

Comment:
you can add a field in the enum to set the number of days instead of having a `toInteger()`.
```dart
enum Days {
  oneday(1),
  twodays(2),
 ...;

  const Days({required this.days});
  final int days;
}
```